src/GitHub.App/Services/ModelService.cs

```diff
@@ -125,8 +126,12 @@ IObservable<IEnumerable<AccountCacheItem>> GetUserOrganizations()
         public IObservable<IReadOnlyList<IRepositoryModel>> GetRepositories()
         {
             return GetUserRepositories(RepositoryType.Owner)
-                .Take(1)
-                .Concat(GetUserRepositories(RepositoryType.Member).Take(1))
+                .PublishLast()
```

Comment:
> Take(1) implies that we only want the first result, so the code that refreshes the cache from the server never runs if there's anything in the cache to return, because it's all run on demand (and we never demand all the results back)

Ah! Ok. That's what you were seeing?



> PublishLast() is an async Last() method, which is what we actually want (to return the most up to date result).

`PublishLast` turns the observable into a connected observable and returns the last notification. And `RefCount` makes a connected observable behave like a normal one. That's why I'm confused by this. There's also a `LastAsync` method to return the last item asynchronously.

But I feel like I'm missing something here in my understanding. :smile: